IOS/SwiftSeedProject/SwiftSeedProject/Application/ViewModels/SourceListViewModel.swift

```diff
@@ -0,0 +1,47 @@
+//
+//  ViewModel.swift
+//  SwiftSeedProject
+//
+//  Created by Brian Sztamfater on 27/3/17.
+//  Copyright © 2017 Making Sense. All rights reserved.
+//
+
+import Bond
+
+class SourceListViewModel {
+    
+    private let sourceService: SourceService!
+    
+    let sources = MutableObservableArray<SourceViewModel>()
+    
+    init(sourceService: SourceService) {
+        self.sourceService = sourceService
+        self.setupNotifications()
+        self.getSources()
```

Comment:
It's usually a bad practice to do stuff other than configuration/setup inside the constructor. Calling `getSources()` and `requestSourcesFromServer()` on the constructor can lead to unwanted behavior.

Constructing an object should fail only on exceptional cases (like trying to initialize it with wrong parameters). Here you have a lot of reasons for the constructor to fail (e.g. CoreData, parsing, network, API), the constructor cannot handle any of those error, and you can't expect consumers to handle such errors when instantiating this ViewModel.

Also, you are hiding behavior. Consumer won't expect this class to do requests to an API upon instantiation.

You could do that more evident by adding an extra parameter on the constructor like `preload` or something like that, but I think it's better to provide methods so consumers can perform those actions on demand.